Fixed Tall Grass placement

SamJBarney opened the pull request:
* Fixed tall grass trying to be placed in Ice Plains, Cold Beach, and Ice Mountains biomes.
* Fixes https://github.com/cuberite/cuberite/issues/2265

NiLSPACE commented on src/Generating/FinishGen.cpp:
```diff
@@ -337,7 +340,7 @@ void cFinishGenTallGrass::GenFinish(cChunkDesc & a_ChunkDesc)
 				a_ChunkDesc.SetBlockTypeMeta(x, y, z, E_BLOCK_TALL_GRASS, 2);
 			}
-			else
+			else if ((biome != biIcePlains) && (biome != biColdBeach) && (biome != biIceMountains) && (biome != biIcePlainsSpikes))
```
I believe we have a `IsBiomeVeryCold` function for this. It's defined in BiomeDef I believe.

NiLSPACE commented on src/Generating/FinishGen.cpp:
```diff
@@ -15,6 +15,7 @@
 #include "../Simulator/FireSimulator.h"
 #include "../World.h"
 #include "../IniFile.h"
+#include "../BiomeDef.h"
```
It *could* be possible that this is also included in the globals. Did you try to compile it without this include?

SamJBarney replied:
Just tested it. Works without it.

NiLSPACE replied:
Then only now a squash can be done, and it can be merged ;)

NiLSPACE commented:
Great that this bug is getting fixed :) It was getting on my nerves, but since I usually avoid snow biomes I didn't see it allot, and thus I forgot about it.

SamJBarney commented:
Is this one ok to merge?

tigerw commented on src/Generating/FinishGen.cpp:
```diff
@@ -322,6 +322,7 @@ void cFinishGenTallGrass::GenFinish(cChunkDesc & a_ChunkDesc)
 			if (
 				(a_ChunkDesc.GetBlockType(x, y, z) != E_BLOCK_AIR) ||
 				((a_ChunkDesc.GetBlockType(x, y - 1, z) != E_BLOCK_GRASS) && (a_ChunkDesc.GetBlockType(x, y - 1, z) != E_BLOCK_DIRT))
+
```
random carriage return is random

tigerw commented:
It presumably works?

SphinxC0re commented:
What do you mean? ... oh :laughing: yeah, squash and rebase onto current master

tigerw commented:
Precisely my dear fellow.

SamJBarney commented:
Squashed it. Sorry, I guess I needed to be more clear with what I meant to ask. I meant to ask if there were any other changes needed before I squashed it.

SphinxC0re commented:
Merge if compiles :smile:

tigerw commented:
>  I meant to ask if there were any other changes needed before I squashed it.

If you do want to fix something, I added a comment on a wild newline appeared above.

SphinxC0re commented:
@tigerw for real? It's a newline.

SamJBarney commented:
A Wild Newline appears!
@tigerw used Scary Face!
The Wild Newline ran away!

----
Sorry. Forgot about that one overnight, tigerw.

SphinxC0re commented:
Oh, @tigerw will push it to the limit! *Baaaaaaaaaaaam!*

tigerw commented:
[Shakes head](http://www.reactiongifs.us/wp-content/uploads/2013/08/shaking_head_breaking_bad.gif)
